crates/ruma-events/src/tag.rs

```diff
@@ -36,29 +41,121 @@ impl From<Tags> for TagEventContent {
     }
 }
 
+/// A user-defined tag name.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct UserTagName {
+    name: String,
+}
+
+impl AsRef<str> for UserTagName {
+    fn as_ref(&self) -> &str {
+        &self.name
+    }
+}
+
+impl FromStr for UserTagName {
+    type Err = &'static str;
```

Comment:
Can you make this return a unit struct that implements `std::error::Error`, i.e.

```rust
pub struct InvalidUserTagName;
```